### PR TITLE
update demo to allow multi-arch/multi-os 

### PR DESCRIPTION
**Pull request by whitneygriffith**
Signed-off-by: Whitney Griffith whitney.griffith16@gmail.com

**Description**: The demo has been explicitly set to run on nodes with Linux OS and and AMD64 Architecture. These changes unblock running the demo on clusters with exclusively linux/arm64 nodes. The node selectors were removed for the demo apps and [node affinity](ttps://kubernetes.io/docs/concepts/configuration/assign-pod-node/) added where necessary. 

**Testing done**: The demo was run locally with these changes. 

**Affected area**:
| Demo                       |  [x] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?

No

2. Is this a breaking change?

No

### CODE REVIEW

**Review by shashankram — CHANGES_REQUESTED** [review]
@whitneygriffith Since these are Linux apps, the node selectors are in place so that they don't get scheduled on Windows nodes in a cluster that has both Linux and Windows nodes.

What's the reason to remove these?

File: demo/deploy-vault.sh
Comment (nojnhuh) [inline demo/deploy-vault.sh:32]: Since this isn't an image we build, let's specify `affinity` here so we can match exactly what the image supports, which `docker manifest inspect vault:1.4.0` shows should be:
- linux/amd64
- linux/arm
- linux/arm64
- linux/386

We do something similar here:
https://github.com/openservicemesh/osm/blob/3a7c924c9ebdedf9513220fc5d1c527b933f71b9/charts/osm/values.yaml#L230-L248

**Review by nojnhuh — COMMENTED** [review]
> @whitneygriffith Since these are Linux apps, the node selectors are in place so that they don't get scheduled on Windows nodes in a cluster that has both Linux and Windows nodes.
> 
> What's the reason to remove these?

The demo apps are currently built and published for Windows, so I'm fairly confident they worked at some point but I haven't tried recently.

I think it's reasonable to remove the node selector for the demo apps so developers are as free as possible to run the demo. If tweaking is necessary for some setups, then the deploy scripts can always be edited manually. These changes unblock running the demo on clusters with exclusively linux/arm64 nodes which I think is at least a step in the right direction.

**Comment by shashankram** [comment]
> > These changes unblock running the demo on clusters with exclusively linux/arm64 nodes which I think is at least a step in the right direction.
> 
> That's fair. @whitneygriffith could you update the commit && PR description to reflect the motivation for this change?

@nojnhuh the images in the demo are not deploying the `-windows` version of the images. So we should still ensure these images don't get scheduled onto Windows nodes. This can be done by specifying a [nodeAffinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity) that allows the pod to be scheduled on either amd64 or arm64 nodes.

**Comment by nojnhuh** [comment]
> @nojnhuh the images in the demo are not deploying the `-windows` version of the images. So we should still ensure these images don't get scheduled onto Windows nodes. This can be done by specifying a [nodeAffinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity) that allows the pod to be scheduled on either amd64 or arm64 nodes.

We don't build separate`-windows` versions of  the demo images anymore. We publish multi-os/arch manifests so the same reference works for both Windows and linux.

**Comment by shashankram** [comment]
> > @nojnhuh the images in the demo are not deploying the `-windows` version of the images. So we should still ensure these images don't get scheduled onto Windows nodes. This can be done by specifying a [nodeAffinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity) that allows the pod to be scheduled on either amd64 or arm64 nodes.
> 
> We don't build separate`-windows` versions of the demo images anymore. We publish multi-os/arch manifests so the same reference works for both Windows and linux.

I see, I forgot about that. Have we tested the multi-arch images to work on Windows? Given that Windows support is still highly experimental, we could potentially not worry about whether the images work on Windows, but it would be good to know rather than hoping it doesn't.

**Comment by nojnhuh** [comment]
I don't remember whether or not those were tested on Windows for that change, and I don't have a Windows setup handy to test them myself.

**Review by shashankram — APPROVED** [review]
Not blocking this on Windows testing.

@whitneygriffith, please describe the motivation for this change in the commit and PR descriptions.

**Comment by whitneygriffith** [comment]
Addressed the comments in last commit

**Review by nojnhuh — APPROVED** [review]

**Pull request merged**
